pkg/stanza/fileconsumer/internal/fingerprint/fingerprint.go

```diff
@@ -18,20 +19,42 @@ const MinSize = 16 // bytes
 // Fingerprint is used to identify a file
 // A file's fingerprint is the first N bytes of the file
 type Fingerprint struct {
-	FirstBytes []byte
+	FirstBytes   []byte
+	HashBytes    uint64
+	BytesUsed    int
+	HashInstance *hash.Hash64
```

Comment:
Since `hash.Hash64` is an interface, I don't think we can meaningfully marshal and unmarshal it. Maybe this should not be exported, and also initialized during fingerprint creation / after unmarshaling.